mesonbuild/coredata.py

```diff
@@ -951,6 +951,8 @@ def get_value(self, key: T.Union[str, OptionKey]):
     def set_value(self, key: T.Union[str, OptionKey], value: T.Union[str, int, bool, T.List[str]]):
         if isinstance(key, str):
             key = OptionKey(key)
+        if self.overrides is None:
+            self.overrides = {}
```

Comment:
Shouldn't we instead be guaranteeing it exists in `__init__`? The idea that it can be None seems odd if we don't expect it to be None.

FWIW you can do this with `dataclasses.field(default_factory=dict)`